pkg/plugins/resources/k8s/store.go

```diff
@@ -273,9 +273,10 @@ func (m *KubernetesMetaAdapter) GetLabels() map[string]string {
 	if _, ok := labels[v1alpha1.DisplayName]; !ok {
 		labels[v1alpha1.DisplayName] = m.GetObjectMeta().GetName()
 	}
-	if m.Namespace != "" {
+	if _, ok := labels[v1alpha1.KubeNamespaceTag]; !ok {
```

Comment:
I assume you want to preserve a namespace from the Zone CP. Is that correct?

I think we still need `if m.Namespace != "" {`. Otherwise, cluster-scoped objects (old policies etc.) will receive a label with an empty value.

Reply:
Yes, that was my idea to preserve the namespace. In case of calling this code on global it was always overriding the namespace and causing that on k8s calling the stats endpoint was incorrect. There was always a request with a namespace of the global sent to the zone.